dotcom-rendering/scripts/islands/island-descriptions.mjs

```diff
@@ -0,0 +1,207 @@
+// @ts-check
+
+import { readdir, readFile, writeFile } from 'node:fs/promises';
+import { resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import rehypeAutolinkHeadings from 'rehype-autolink-headings';
+import rehypeSlug from 'rehype-slug';
+import rehypeStringify from 'rehype-stringify';
+import remarkParse from 'remark-parse';
+import remarkRehype from 'remark-rehype';
+import { unified } from 'unified';
+import * as zod from 'zod';
+
+/** @type {(name: string) => RegExp} */
+const getRegExForIsland = (name) =>
+	new RegExp(
+		'\n/\\*\\*\n?' +
+			'((?: \\*.*\n)+)' +
+			' \\*\\/' +
+			`\nexport const ${name} =`,
+		'm',
+	);
+
+const componentsDirectory = resolve(
+	fileURLToPath(import.meta.url),
+	'../../../src/web/components',
+);
+
+const PREFIX = 'window.chartData = ';
+
+const chunk = zod.object({
+	label: zod.string(),
+	isAsset: zod.boolean(),
+	statSize: zod.number(),
+	parsedSize: zod.number(),
+	gzipSize: zod.number(),
+});
+
+/** Sorted by gzipSize */
+const getBundleReport = () =>
+	readFile(
+		resolve(
+			fileURLToPath(import.meta.url),
+			'../../../dist/browser.modern-bundles.html',
+		),
+		'utf-8',
+	)
+		.then((file) =>
+			file
+				.split('\n')
+				.find((line) => line.trim().startsWith(PREFIX))
```

Comment:
I think we can avoid this ... awkwardness ... by using the `generateStatsFile` option for webpack bundle analyser? https://github.com/webpack-contrib/webpack-bundle-analyzer#options-for-plugin

Reply:
is that file a problem? how much of an impact does it have?

Reply:
I think that the server toppled over last year from having too many files on disk, hence my reservation, as it’s Webpack’s full bundle JSON, which does not include gzipped sizes.

I could try running the WebpackAnalyser plugin twice, once with the `html` output and once with the `json` output, possibly?